django/utils/inspect.py

```diff
@@ -3,14 +3,23 @@
 
 
 @functools.lru_cache(maxsize=512)
-def _get_signature(func):
-    return inspect.signature(func)
+def _get_func_parameters(func, remove_self):
+    parameters = tuple(inspect.signature(func).parameters.values())
+    if remove_self:
+        parameters = parameters[1:]
```

Comment:
@adamchainz 
It says to remove self but can it get called for a static method or a class method?
In the case of classmethod it will remove the cls. In the case of staticmethod it may remove argument.

Reply:
Not an issue for `staticmethod` has they are not _methods_ per-se they are mainly just a flag to let Python type building know that the method should not be bound. In other words

```python
class Foo:
    @staticmethod
    def static_bar(arg): pass

inspect.ismethod(Foo.static_bar)  # False
```

You point holds for `classmethod` though but the code works as expected: it's a method so the first argument is ignored if it's bound to an object. We could likely just rename the `remove_self` parameter to `remove_first` to avoid confusion but we certainly add tests to `@classmethod`.